In [1]:
# Importing the necessary libraries
import pandas as pd
import numpy as np

import pymysql # works as my sql adaptor
from sqlalchemy import create_engine


In [2]:
df = pd.read_csv('Walmart.csv')

df.shape

(10051, 11)

### There are currently 10051 rows and 11 columns in the dataset before cleanig 

In [3]:
df.head()

,invoice_id,Branch,City,category,unit_price,quantity,date,time,payment_method,rating,profit_margin
0,1,WALM003,San Antonio,Health and beauty,$74.69,7.0,05/01/19,13:08:00,Ewallet,9.1,0.48
1,2,WALM048,Harlingen,Electronic accessories,$15.28,5.0,08/03/19,10:29:00,Cash,9.6,0.48
2,3,WALM067,Haltom City,Home and lifestyle,$46.33,7.0,03/03/19,13:23:00,Credit card,7.4,0.33
3,4,WALM064,Bedford,Health and beauty,$58.22,8.0,27/01/19,20:33:00,Ewallet,8.4,0.33
4,5,WALM013,Irving,Sports and travel,$86.31,7.0,08/02/19,10:37:00,Ewallet,5.3,0.48


In [4]:
df.describe()


,invoice_id,quantity,rating,profit_margin
count,10051.000000,10020.000000,10051.000000,10051.000000
mean,5025.741220,2.353493,5.825659,0.393791
std,2901.174372,1.602658,1.763991,0.090669
min,1.000000,1.000000,3.000000,0.180000
25%,2513.500000,1.000000,4.000000,0.330000
50%,5026.000000,2.000000,6.000000,0.330000
75%,7538.500000,3.000000,7.000000,0.480000
max,10000.000000,10.000000,10.000000,0.570000


In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10051 entries, 0 to 10050
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   invoice_id      10051 non-null  int64  
 1   Branch          10051 non-null  object 
 2   City            10051 non-null  object 
 3   category        10051 non-null  object 
 4   unit_price      10020 non-null  object 
 5   quantity        10020 non-null  float64
 6   date            10051 non-null  object 
 7   time            10051 non-null  object 
 8   payment_method  10051 non-null  object 
 9   rating          10051 non-null  float64
 10  profit_margin   10051 non-null  float64
dtypes: float64(3), int64(1), object(7)
memory usage: 863.9+ KB


In [6]:
# Checking Duplicates records
df.duplicated().sum()

np.int64(51)

### There are 51 duplicates in the dataset so we are Removing them

In [7]:
df.drop_duplicates(inplace=True)

In [8]:
# Checking for null values
df.isnull().sum()

invoice_id         0
Branch             0
City               0
category           0
unit_price        31
quantity          31
date               0
time               0
payment_method     0
rating             0
profit_margin      0
dtype: int64

### There are 31 Null values in both unit_price and quantity columns so we are removing the null records

In [9]:
df.dropna(inplace=True)

In [10]:
df.shape

(9969, 11)

###  After removing the null values there are 9969 rows and 11 columns in the dataset

In [11]:
# checking the data types of the columns
df.dtypes

invoice_id          int64
Branch             object
City               object
category           object
unit_price         object
quantity          float64
date               object
time               object
payment_method     object
rating            float64
profit_margin     float64
dtype: object

In [12]:
# changing the data type of the unit_price columns to float and removing the $ sign

df['unit_price'] = df['unit_price'].str.replace("$", "").astype(float)

In [13]:
df.head()

,invoice_id,Branch,City,category,unit_price,quantity,date,time,payment_method,rating,profit_margin
0,1,WALM003,San Antonio,Health and beauty,74.69,7.0,05/01/19,13:08:00,Ewallet,9.1,0.48
1,2,WALM048,Harlingen,Electronic accessories,15.28,5.0,08/03/19,10:29:00,Cash,9.6,0.48
2,3,WALM067,Haltom City,Home and lifestyle,46.33,7.0,03/03/19,13:23:00,Credit card,7.4,0.33
3,4,WALM064,Bedford,Health and beauty,58.22,8.0,27/01/19,20:33:00,Ewallet,8.4,0.33
4,5,WALM013,Irving,Sports and travel,86.31,7.0,08/02/19,10:37:00,Ewallet,5.3,0.48


In [14]:
df.columns

Index(['invoice_id', 'Branch', 'City', 'category', 'unit_price', 'quantity',
       'date', 'time', 'payment_method', 'rating', 'profit_margin'],
      dtype='object')

In [21]:
# Coverting the column names to lower case
df. columns = df. columns.str.lower()

In [15]:
# Creating new Feature to calculate the total sales
df['Total_sales'] = df['unit_price'] * df['quantity']
df.head()

,invoice_id,Branch,City,category,unit_price,quantity,date,time,payment_method,rating,profit_margin,Total_sales
0,1,WALM003,San Antonio,Health and beauty,74.69,7.0,05/01/19,13:08:00,Ewallet,9.1,0.48,522.83
1,2,WALM048,Harlingen,Electronic accessories,15.28,5.0,08/03/19,10:29:00,Cash,9.6,0.48,76.40
2,3,WALM067,Haltom City,Home and lifestyle,46.33,7.0,03/03/19,13:23:00,Credit card,7.4,0.33,324.31
3,4,WALM064,Bedford,Health and beauty,58.22,8.0,27/01/19,20:33:00,Ewallet,8.4,0.33,465.76
4,5,WALM013,Irving,Sports and travel,86.31,7.0,08/02/19,10:37:00,Ewallet,5.3,0.48,604.17


### New Column Created to calculate the total sales (Unit_price * Quantity)

In [16]:
df.to_csv('Walmart_cleaned.csv', index=False)

In [17]:
df.to_sql

<bound method NDFrame.to_sql of       invoice_id   Branch         City                category  unit_price  \
0              1  WALM003  San Antonio       Health and beauty       74.69   
1              2  WALM048    Harlingen  Electronic accessories       15.28   
2              3  WALM067  Haltom City      Home and lifestyle       46.33   
3              4  WALM064      Bedford       Health and beauty       58.22   
4              5  WALM013       Irving       Sports and travel       86.31   
...          ...      ...          ...                     ...         ...   
9995        9996  WALM056      Rowlett     Fashion accessories       37.00   
9996        9997  WALM030   Richardson      Home and lifestyle       58.00   
9997        9998  WALM050     Victoria     Fashion accessories       52.00   
9998        9999  WALM032        Tyler      Home and lifestyle       79.00   
9999       10000  WALM069     Rockwall     Fashion accessories       62.00   

      quantity      date      t

In [18]:
# host = "localhost"
# port = 3306
# user = "root"
# password = "anirudha"
# database = "walmart_sales"

In [19]:
# mysql Connection
engine = create_engine("mysql+pymysql://root:anirudha@localhost/walmart_sales")

try :
    engine
    print("Connection Successful")
except :
    print("Connection Failed")

Connection Successful


In [22]:
df.to_sql(name='walmart_sales', con=engine, if_exists='append', index=False)

9969